src/libraries/System.Text.Json/src/System/Text/Json/Reader/JsonReaderHelper.Unescaping.cs

```diff
@@ -159,6 +159,50 @@ public static bool UnescapeAndCompare(ReadOnlySequence<byte> utf8Source, ReadOnl
             return result;
         }
 
+        public static bool UnescapeAndCompareBothInputs(ReadOnlySpan<byte> utf8Source1, ReadOnlySpan<byte> utf8Source2)
+        {
+            int index1 = utf8Source1.IndexOf(JsonConstants.BackSlash);
+            int index2 = utf8Source2.IndexOf(JsonConstants.BackSlash);
+
+            Debug.Assert(index1 >= 0, "the first parameter is not escaped");
+            Debug.Assert(index2 >= 0, "the second parameter is not escaped");
+
+            byte[]? unescapedArray1 = null;
+            byte[]? unescapedArray2 = null;
+
+            Span<byte> utf8Unescaped1 = utf8Source1.Length <= JsonConstants.StackallocByteThreshold ?
+                stackalloc byte[JsonConstants.StackallocByteThreshold] :
+                (unescapedArray1 = ArrayPool<byte>.Shared.Rent(utf8Source1.Length));
+
+            Span<byte> utf8Unescaped2 = utf8Source2.Length <= JsonConstants.StackallocByteThreshold ?
```

Comment:
This seems to assume that unescaping never increases length?

Reply:
Unescaping should always decrease the length. I've copied this logic from the other unescaping routines in the same file.